elasticdl/python/worker/worker.py

```diff
@@ -469,6 +504,22 @@ def run(self):
             mode = Mode.EVALUATION
         else:
             mode = Mode.TRAINING
+
+        # need to get model from PS, i.e. `train_with_local_model=False`
+        # at following situations:
+        #     processing first minibatch
+        #     last minibatch is evaluation task
+        #     last minibatch is training task and failed
+        #     local_update_count >= worker._get_model_steps
+        # otherwise, train with local model, i.e. `train_with_local_model=True`
+        train_with_local_model = False
+
+        local_update_count = 0
+        last_training_minibatch_failed = False
+        # initialize `last_minibatch_do_evaluation` to True in order to set
+        # `train_with_local_model` to False inside forloop for the first
+        # minibatch.
+        last_minibatch_do_evaluation = True
```

Comment:
Is it better to set:
`local_update_count == self._get_model_steps`  instead?

Reply:
Done.